WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubTimeRangeController.swift

```diff
@@ -0,0 +1,146 @@
+import Foundation
+import Yosemite
+
+struct TimeRange {
+    let start: Date
+    let end: Date
+}
+
+public class AnalyticsHubTimeRangeController {
+
+    private let currentTimezone = TimeZone.current
+    private let currentDate: Date
+    private let currentCalendar: Calendar
+    private let selectionType: SelectionType
+
+    lazy private(set) var currentTimeRange: TimeRange = {
+        generateCurrentTimeRangeFrom(selectionType: selectionType)
+    }()
+
+    lazy private(set) var previousTimeRange: TimeRange = {
+        generatePreviousTimeRangeFrom(selectionType: selectionType)
+    }()
+
+    lazy private(set) var currentRangeDescription: String = {
+        generateDescriptionOf(timeRange: currentTimeRange)
+    }()
+
+    lazy private(set) var previousRangeDescription: String = {
+        generateDescriptionOf(timeRange: previousTimeRange)
+    }()
+
+    var selectionDescription: String {
+        selectionType.description
+    }
+
+    init(selectedTimeRange: StatsTimeRangeV4,
+         currentDate: Date = Date(),
+         currentCalendar: Calendar = Calendar(identifier: .iso8601)
```

Comment:
Can you explain to me a little bit about why we use this specific calendar?

Reply:
As discussed in this PR, I've added a TODO to adjust this usage later. https://github.com/woocommerce/woocommerce-ios/pull/8193/commits/f3666968c4000757f8f027af4c44fc45909138aa